src/components/App/App.tsx

```diff
@@ -29,6 +35,12 @@ export default class App extends React.Component<{}, AppState> {
             selectedSorting: SortOption.ReleaseDate,
             movies,
             selectedMovieId: null,
+            showModal: true,
+
+            // change isEditMovieModal/isAddMovieModal/isDeleteMovieModal to see use cases
+            isEditMovieModal: true,
```

Comment:
Optional. it is better in such cases not to add Boolean for each type of dialog. It is better to have one state variable. For example movieModalType.
For this in TS we even can use typing like type ModalType = 'Edit' | 'Delete' | 'Add'